bodhi/server/models.py

```diff
@@ -2056,24 +2056,48 @@ def check_requirements(self, session, settings):
             return False, "Failed to determine last_modified: %r" % e.message
 
         try:
-            query = dict(title=self.title, since=since)
+            # query results for this update
+            query = dict(type='bodhi_update', item=self.alias, since=since,
+                         testcases=','.join(requirements))
             results = bodhi.server.util.taskotron_results(settings, **query)
-        except IOError as e:
-            return False, "Failed to talk to taskotron: %r" % e.message
+
+            # query results for each build
+            # retrieve timestamp for each build so that queries can be optimized
+            koji = buildsys.get_session()
+            koji.multicall = True
+            for build in self.builds:
+                koji.getBuild(build.nvr)
+            buildinfos = koji.multiCall()
+
+            for index, build in enumerate(self.builds):
+                buildinfo = buildinfos[index]
+                if not isinstance(buildinfo, dict):
+                    log.warn("Koji build info doesn't seem correct: %r",
+                             buildinfo)
```

Comment:
It will fail on the next step, probably raising `TypeError`. Do you think I should raise that immediately?